components/AppLayout/AppHeader.js

```diff
@@ -232,7 +232,7 @@ function AppHeader({
             <Links classes={classes} unsolvedCount={unsolvedCount} />
           </Box>
         </div>
-        <GlobalSearch onIconClick={() => setDisplayLogo(false)} />
```

Comment:
Personally I would prefer `onExpanded` or `onToggle` because
this is the scenario that a child component component notifies 

```js
 <GlobalSearch onExpanded={expanded => setDisplayLogo(!expanded)} />
```

Reply:
I've refactor the relation between the parent and the child component so that parent listen to child's change rather then the child change the parent directly